suite/chaos/reorg_check.go

```diff
@@ -0,0 +1,205 @@
+package chaos
+
+import (
+	"context"
+	"fmt"
+	"github.com/avast/retry-go"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/pkg/errors"
+	"github.com/rs/zerolog/log"
+	"github.com/smartcontractkit/integrations-framework/chaos/experiments"
+	"github.com/smartcontractkit/integrations-framework/client"
+	"github.com/smartcontractkit/integrations-framework/config"
+	"github.com/smartcontractkit/integrations-framework/environment"
+	"golang.org/x/sync/errgroup"
+	"math/big"
+	"sort"
+	"strings"
+	"sync"
+	"time"
+)
+
+const (
+	MinerIDTemplate    = "miner-%d"
+	ReorgCheckAttempts = 15
+	ReorgCheckInterval = 10 * time.Second
+)
+
+// ReorgChecker collects data to verify reorged blocks
+type ReorgChecker struct {
```

Comment:
As per my above comment, I feel this object can be converted to a header receiver for the ETH client. It fits the pattern.